syntax = "proto3";

service DocumentService {
    rpc UploadDocument (UploadRequest) returns (UploadResponse);
    rpc SearchDocuments (SearchRequest) returns (SearchResponse);
    rpc SummarizeText (SummarizeRequest) returns (SummarizeResponse);
}

message UploadRequest {
    bytes file_content = 1;
    string document_name=2;
}

message UploadResponse {
    string document_id = 1;
}

message SearchRequest {
    string query = 1;
}

message SearchResponse {
    // Your search response message
    repeated string document_ids = 1;
    repeated string document_summaries = 2;
    repeated string document_texts = 3;
}

message SummarizeRequest {
    // Your text summarization request message
    string text = 1;
}

message SummarizeResponse {
    // Your text summarization response message
    string summary = 1;
}
